### ISSUE TITLE
concurrent requests cause configcache errors (windows-dev mode)

### ISSUE THREAD

**Issue by lekoala** [silverstripe/silverstripe-framework#10634]
## Affected Version

4.*
dev only
windows os

## Description

Due to concurrent requests and cache not working very well on windows, it happens that junk output is added due to

https://github.com/silverstripe/silverstripe-framework/blob/dcdc25500be729f4340d42f8a4fc797461f862f3/src/Core/Cache/ManifestCacheFactory.php#L26-L30

The following output is added at the end:
manifestcache-log.WARNING: Failed to save values: rename(...\silverstripe-cache\...\configcache\1d1131f576b7,...\silverstripe-cache\...\configcache\U\F\QAdjY48zDlbAQWHpYUrw): Access is denied (code: 5) : Access is denied (code: 5) at ...\\vendor\\symphony\\cache\\Traits\\FilesystemCommonTrait.php:115)"}

this breaks ajax requests for example

Related:
https://github.com/silverstripe/silverstripe-graphql/issues/257

## Steps to Reproduce

Visit a page in the admin with two gridfields
two requests for SearchForm schema start at the same time
the second one get junk output due to manifestcache-log

## Proposal

Do not output to stream

Or at least have some type of mechanism to configure this easily, for example

here I've reused the SS_ERROR_LOG configuration which makes sense so that you only look in one place for errors.

```php
        if (!$logger) {
            $logger = new Logger("manifestcache-log");
            $errorLog = Environment::getEnv('SS_ERROR_LOG');
            if ($errorLog) {
                $basePath = BASE_PATH; // don't use Director::baseFolder() since manifest is not built yet
                $logger->pushHandler(new StreamHandler($basePath . '/' . $errorLog, Logger::WARNING));
            } else {
                if (Director::isDev()) {
                    $logger->pushHandler(new StreamHandler('php://output'));
                } else {
                    $logger->pushHandler(new ErrorLogHandler());
                }
            }
        }
```

**Comment by lozcalver** [comment]
Are you using PHP 8.1? Could be related to this: https://github.com/php/php-src/issues/7910

**Comment by lekoala** [comment]
> Are you using PHP 8.1? Could be related to this: [php/php-src#7910](https://github.com/php/php-src/issues/7910)

@kinglozzer yes i am ! never thought it could be a php issue (I first thought it was xdebug). Well, I guess it's going to be fixed some day then :-)
Nevertheless, I don't think outputting blindly to php output works well when there are ajax requests / api requests. In fact I think that's only valid in cli context. I quite like outputting to error log in other cases.

**Comment by GuySartorelli** [comment]
I'm going to mark this as an enhancement, since the actual buggy behaviour is due to a PHP bug, but you're right that it would be beneficial to be able to configure this.

**Comment by lekoala** [comment]
i created an issue on the symfony repository, the solution seems to be : do not use the rename function ^^

**Comment by elkebe** [comment]
I'm also having this issue. In the interim, how do I actually clear the junk error so it doesn't display on every page in dev mode?
